src/topology/vector_bundle.lean

```diff
@@ -387,3 +404,102 @@ lemma is_open_map_proj : is_open_map Z.proj :=
 topological_fiber_bundle_core.is_open_map_proj Z
 
 end topological_vector_bundle_core
+
+end
+
+section
+
+/-! ### Topological vector prebundle -/
+
+variable [∀ x, topological_space (E x)]
+
+open topological_space
+
+/-- This structure permits to define a vector bundle when trivializations are given as local
+equivalences but there is not yet a topology on the total space. The total space is hence given a
+topology in such a way that there is a fiber bundle structure for which the local equivalences
+are also local homeomorphism and hence vector bundle trivializations. -/
+@[nolint has_inhabited_instance]
+structure topological_vector_prebundle :=
+(trivialization_at : B → topological_vector_prebundle.trivialization R F E)
+(mem_base_trivialization_at : ∀ x : B, x ∈ (trivialization_at x).base_set)
+(continuous_triv_change : ∀ x y : B, continuous_on ((trivialization_at x) ∘
+  (trivialization_at y).to_local_equiv.symm) ((trivialization_at y).target ∩
+  ((trivialization_at y).to_local_equiv.symm ⁻¹' (trivialization_at x).source)))
+(total_space_mk_inducing : ∀ (b : B), inducing ((trivialization_at b) ∘ (total_space_mk E b)))
+
+namespace topological_vector_prebundle
+
+variables {R E F}
+
+/-- Natural identification as a `topological_fiber_prebundle`. -/
+def to_topological_fiber_prebundle (a : topological_vector_prebundle R F E) :
+  topological_fiber_prebundle F (proj E) :=
+{ trivialization_at := λ x, a.trivialization_at x, ..a }
+
+/-- Topology on the total space that will make the prebundle into a bundle. -/
+def total_space_topology (a : topological_vector_prebundle R F E) :
+  topological_space (total_space E) :=
+a.to_topological_fiber_prebundle.total_space_topology
+
+/-- Promotion from a `topologial_vector_prebundle.trivialization` to a
+  `topologial_vector_bundle.trivialization`. -/
+def vector_bundle_trivialization_at (a : topological_vector_prebundle R F E) (x : B) :
+  @topological_vector_bundle.trivialization R _ F E _ _ _ _ _ _ _ a.total_space_topology  :=
+begin
+  letI := a.total_space_topology,
+  exact { linear := (a.trivialization_at x).linear,
+  ..a.to_topological_fiber_prebundle.bundle_trivialization_at x }
+end
+
+variable (a : topological_vector_prebundle R F E)
+
+lemma mem_trivialization_at_source (b : B) (x : E b) :
+  total_space_mk E b x ∈ (a.trivialization_at b).source :=
+begin
+  simp only [(a.trivialization_at b).to_prebundle_trivialization.source_eq, mem_preimage, proj],
+  exact a.mem_base_trivialization_at b,
+end
+
+lemma total_space_mk_preimage_soure (b : B) :
+  (total_space_mk E b) ⁻¹' (a.trivialization_at b).source = univ :=
+begin
+  apply eq_univ_of_univ_subset,
+  rw [(a.trivialization_at b).source_eq, ←preimage_comp, function.comp],
+  simp only [proj],
```

Comment:
I marked `proj` as reducible so why do I need to do this?

Reply:
I am not sure `reducible` does what you think it does.

Reply:
`total_space_mk` is marked as reducible and I do not need to rewrite it here, so I thought that it meant that it was treated as notation. I am just surprised by the different behavior of `total_space_mk` and `proj`.

Reply:
@eric-wieser you are right that `rw` seems to see through reducible defs, but it does not work here for `proj` even though it works for `total_space_mk` (as in if I take out the reducible from it then the rw here does not work)!! Can you understand why?